mllib/src/main/scala/org/apache/spark/mllib/feature/IDF.scala

```diff
@@ -218,7 +218,7 @@ private object IDFModel {
           newValues(k) = values(k) * idf(indices(k))
           k += 1
         }
-        Vectors.sparse(n, indices, newValues)
+        Vectors.sparse(n, indices, newValues).toSparse
```

Comment:
Hm, that's already a fairly heavy operation, so I tend to think it's OK to filter out explicit zeroes. @jkbradley am I missing anything? seems like a win. I'm trying to think of why any code would legitimately rely on an explicit 0.